infoTree/createRecycleTree.js

```diff
@@ -0,0 +1,208 @@
+  }
+  var newline = new Node('|');
+  var queue = [this.root, newline];
+  var string = '';
+  while(queue.length) {
+    var node = queue.shift();
+    string += node.data.toString() + ' ';
+    if(node === newline && queue.length) {
+      queue.push(newline);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  console.log(string.slice(0, -2).trim());
+};
+Tree.prototype.printByLevel = function() {
+  if(!this.root) {
+    return console.log('No root node found');
+  }
+  var newline = new Node('\n');
+  var queue = [this.root, newline];
+  var string = '';
+  while(queue.length) {
+    var node = queue.shift();
+    string += node.data.toString() + (node.data !== '\n' ? ' ' : '');
+    if(node === newline && queue.length) {
+      queue.push(newline);
+    }
+    for(var i = 0; i < node.children.length; i++) {
+      queue.push(node.children[i]);
+    }
+  }
+  console.log(string.trim());
+};
+//TODO: takes in a leaf data string and returns it's address as a string formatted [num, num, ... num]
+//where tree.root.children[num].children[num]...children[num].data is the string put in
+Tree.prototype.findAddress = function(data){
+	if(!this.contains(data, this.traverseBF)){
+		return console.log("Tree does not contain "+ data);
+	}
+	return true
+}
+
+Tree.prototype.findEveryLeaf = function(){
+	var leafArray = []
+	//TODO: find every leaf Node and push it into leafArray
+	return leafArray;
+}
+
+//finds the URL to scrape data from by adding parent's name to base URL
+Tree.prototype.findUrl = function(leaf){
+	url = "https://www.seattle.gov/util/MyServices/WhereDoesItGo/"
+	while(leaf.data!=='rootNode'){
+		url += leaf.data + "/";
+		//Bad Idea. Will fix
+		leaf = leaf.parent;
+	}
+	url += "index.htm";
+	return url
+}
+// End Tree Implementation
+
+
+
+
+
+
+
+
+
+
+
+//formatting whereDoesItGo. Changes the value string => array. 
+//This could be done with regular expressions, but an array containing the parents seems easier to me.
+//TODO: adapt for different inputs. because of the slice(4,-1) this could break easily when we include the category titles
+function formatRecycleArray(whereRecycleArray){
+  whereRecycleArray.forEach(branchInfo => branchInfo.value = branchInfo.value.split("/").slice(4,-1))
+  return whereRecycleArray;
+}
+
+//creates a tree with a root node of "rootNode" with children of general recycle materials based on "Where does it go" on seattle.gov
+//each of those branches can have more branches with more specific recycling materials/objects
+export default function createRecycleTree(recycleArray){
+  var recycleTree = new Tree()
+  recycleTree.add("rootNode")
+
+  //format object so its easier to use if it's not formatted
+  (typeof recycleArray[0].value==="string") && (recycleArray = formatRecycleArray(recycleArray))
+
+  recycleArray.forEach(branchInfo => {
+    branchInfo.value.forEach((branchName, index) => 
+      !recycleTree.contains(branchName) && 
```

Comment:
this will be easier to read if you pull it out to a filter
```
branchInfo.value.filter( branchName => recycleTree.contains(branchName)).forEach((branchName, index) =>  
```

Reply:
ah, I missed that this is a dedupe. What happens if you try to add a duplicate? Does it overwrite it? When would there be duplicates?

Reply:
If you add a duplicate, it makes another node in the tree. We could add an internal structure in the tree to protect against duplicate branches, but It's possible you run into an issue with similar terms (cans, bottles, box, etc. can all be made of different materials). There are many duplicates since you're adding the 2nd level branches each time the forEach runs over them. And you can't exclude them, since the leaf is the information you're looking for